### PR TITLE
Upgrade integration tests to use AGP 7.3/Gradle 7.4

### PR DESCRIPTION
**Pull request by gmackall**
Upgrades tests under `dev/integration_tests` to use the latest values [in the templates](https://github.com/flutter/flutter/blob/944d6c8fef76b2db7a2c33eeb1653de2b8571fd8/dev/tools/bin/generate_gradle_lockfiles.dart#L135). Most of this PR was auto-generated, by running `find dev/integration_tests/ -type d -name 'android' | dart dev/tools/bin/generate_gradle_lockfiles.dart` from the root of the flutter directory.

The pieces that were not are:
1. Upgrading the Gradle versions used in integration tests to be >=7.4, in places where it was currently lower.
2. Upgrading the mac, windows, and linux build_tests .ci.yaml configuration to use jdk 17 on all shards. It currently was using a mix of 17 and 11. This isn't desirable, because some of the tests require 17, and the distribution is [random across shards](https://github.com/flutter/flutter/blob/cef9a8c54b766fa5f66e6fbdf7df341176f8e29f/dev/bots/test.dart#L553) (so they were only passing because they were getting randomly placed on shards using jdk 17).
3. Adding a dependency on jdk 11 for the tests based on the `firebaselab/firebaselab` recipe. Previously they had no java dependency, and were therefore defaulting to java 1.8. The newer AGP versions necessitated an upgrade to 11 to run.

Note that it also ended up downgrading the AGP version in two places (in the [hybrid_android_views](https://github.com/flutter/flutter/pull/129642/files#diff-d7cfaa1c96d5c2383e7fc253ac8c9df062b544c24f7d80aa8a02290c01d17205L17) and [platform_interaction](https://github.com/flutter/flutter/pull/129642/files#diff-b2fba811efadd1cfb097bebd93be99fe4b6d6f17aefa863267290fc704290766L17) tests), because those had been manually edited to a newer version than the template.

Related to: https://github.com/flutter/flutter/issues/123636, https://github.com/flutter/flutter/pull/123910

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by gmackall** [comment]
This may also unblock https://github.com/flutter/flutter/pull/122779 (see [this comment](https://github.com/flutter/flutter/pull/122779)), though the changes are worth landing independently.

**Review by camsim99 — APPROVED** [review]
LGTM! Can you note in the description where AGP was downgraded? I assume it's not a problem but may be good to remember.

**Comment by gmackall** [comment]
> LGTM! Can you note in the description where AGP was downgraded? I assume it's not a problem but may be good to remember.

Updated the description to link to the two places

**Comment by gmackall** [comment]
As a follow up it might be good to make the shards these tests use for a specific platform (e.g. Mac build_tests_1_4,...,Mac build_tests_4_4) share a common configuration, or at least have a comment noting changes should be made to all shards. 

Otherwise the test behavior is implicitly relying on the current distribution of tests across the shards, which makes it easy to introduce issues like this where failures only manifest on later, unrelated PRs.

File: dev/integration_tests/android_embedding_v2_smoke_test/android/build.gradle
Comment (johnmccutchan) [inline dev/integration_tests/android_embedding_v2_smoke_test/android/build.gradle:10]: Is it possible and should we try and use named constants for these version numbers (here and elsewhere)?

**Review by johnmccutchan — APPROVED** [review]

File: dev/integration_tests/android_embedding_v2_smoke_test/android/build.gradle
Comment (gmackall) [reply dev/integration_tests/android_embedding_v2_smoke_test/android/build.gradle:10]: From testing it looks like constants in the build.gradle get resolved to their values in the lockfiles, so if we did use a constant in the generated `build.gradle`, it would still resolve to a specific value in the lockfile that wouldn't change in line with updates to the constant. So we would still need to re-run the script, which replaces the value in the generated `build.gradle` anyways. 

I do think it could make sense to have the values in the `generate_gradle_lockfiles.dart` templates be constants, especially if we did want them to be the same as the values in templates we use to make a new project. I'll file an issue where we can make the decision.

**Comment by gmackall** [comment]
It looks like previous versions of similar PRs have had a tendency to break in postsubmit (see https://github.com/flutter/flutter/commit/d155bc1bad4d069e72eb93aa7e46f127b0d92356, https://github.com/flutter/flutter/commit/1e69683688cc107f3e02c259a79453b9e30a200f, https://github.com/flutter/flutter/commit/716533925b5c0c2f2f583db1653d6d0013259adb). 

cc'ing @christopherfujino as I see you've reviewed some of these past PRs - I tried the devicelab tests that looked relevant to the tests I changed [as outlined here](https://github.com/flutter/flutter/tree/96a2c05358106372de53e1d3b1a397e3a5589282/dev/devicelab#how-the-devicelab-runs-tests), and I didn't run into any issues. Is there anything else you would recommend trying before merging? There still isn't a good way to run postsubmits before submitting, is there?

File: dev/integration_tests/android_embedding_v2_smoke_test/android/build.gradle
Comment (johnmccutchan) [reply dev/integration_tests/android_embedding_v2_smoke_test/android/build.gradle:10]: SGTM.

**Comment by christopherfujino** [comment]
> It looks like previous versions of similar PRs have had a tendency to break in postsubmit (see [d155bc1](https://github.com/flutter/flutter/commit/d155bc1bad4d069e72eb93aa7e46f127b0d92356), [1e69683](https://github.com/flutter/flutter/commit/1e69683688cc107f3e02c259a79453b9e30a200f), [7165339](https://github.com/flutter/flutter/commit/716533925b5c0c2f2f583db1653d6d0013259adb)).
> 
> cc'ing @christopherfujino as I see you've reviewed some of these past PRs - I tried running the devicelab tests that looked relevant to the tests I changed [as outlined here](https://github.com/flutter/flutter/tree/96a2c05358106372de53e1d3b1a397e3a5589282/dev/devicelab#how-the-devicelab-runs-tests), and I didn't run into any issues. Is there anything else you would recommend trying before merging? There still isn't a good way to run postsubmits before submitting, is there?

You can test devicelab targets by temporarily changing `presubmit: false` to true in https://github.com/flutter/flutter/blob/master/.ci.yaml#L1593. Just remember to revert the changes before submitting :)

**Comment by gmackall** [comment]
> > It looks like previous versions of similar PRs have had a tendency to break in postsubmit (see [d155bc1](https://github.com/flutter/flutter/commit/d155bc1bad4d069e72eb93aa7e46f127b0d92356), [1e69683](https://github.com/flutter/flutter/commit/1e69683688cc107f3e02c259a79453b9e30a200f), [7165339](https://github.com/flutter/flutter/commit/716533925b5c0c2f2f583db1653d6d0013259adb)).
> > cc'ing @christopherfujino as I see you've reviewed some of these past PRs - I tried running the devicelab tests that looked relevant to the tests I changed [as outlined here](https://github.com/flutter/flutter/tree/96a2c05358106372de53e1d3b1a397e3a5589282/dev/devicelab#how-the-devicelab-runs-tests), and I didn't run into any issues. Is there anything else you would recommend trying before merging? There still isn't a good way to run postsubmits before submitting, is there?
> 
> You can test devicelab targets by temporarily changing `presubmit: false` to true in https://github.com/flutter/flutter/blob/master/.ci.yaml#L1593. Just remember to revert the changes before submitting :)

Awesome, thanks I'll try that!

**Comment by gmackall** [comment]
There were some failures in enabling the post-submits, but I checked the logs for each one and they all seemed infra-related (they never started for some reason).

Most passed though, and nothing indicated a problem with this change, so I'm going to go ahead and add autosubmit.

**Comment by christopherfujino** [comment]
> auto label is removed for flutter/flutter, pr: 129642, due to - The status or check suite [Mac customer_testing](https://ci.chromium.org/b/8776846535035351073) has failed. Please fix the issues identified (or deflake) before re-applying this label.

I'm working on fixing this upstream. You will have to rebase (not merge) upstream once the tree is fixed to fix the tests on your branch.

**Comment by gmackall** [comment]
It looks like this caused two runs of the `Mac build_tests_2_4` to fail for earlier PRs, because they ended up starting after ([first](https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20build_tests_2_4/11541/overview), [second](https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20build_tests_2_4/11540/overview)) the changes landed, and I assume are pulling the latest .ci.yaml when the test gets started? That must be what is happening, because the dependencies section shows them using jdk 17, which is a change from this PR. Surprising, I would expect the .ci.yaml changes to only start taking effect in the builds for the current commit, not earlier.

Either way it shouldn't cause problems for future commits, though it may affect some presubmits until people merge or rebase.

**Comment by christopherfujino** [comment]
> It looks like this caused two runs of the `Mac build_tests_2_4` to fail for earlier PRs, because they ended up starting after ([first](https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20build_tests_2_4/11541/overview), [second](https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20build_tests_2_4/11540/overview)) the changes landed, and I assume are pulling the latest .ci.yaml when the test gets started? That must be what is happening, because the dependencies section shows them using jdk 17, which is a change from this PR. Surprising, I would expect the .ci.yaml changes to only start taking effect in the builds for the current commit, not earlier.
> 
> Either way it shouldn't cause problems for future commits, though it may affect some presubmits until people merge or rebase.

It's actually that these configurations are generated by a post-submit job, so there's always some lag between when the PR lands and when they take effect: https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20ci_yaml%20flutter%20roller/9809/overview

**Comment by christopherfujino** [comment]
Oh, and it looks like these ran out of order https://ci.chromium.org/ui/p/flutter/builders/luci.flutter.prod/Linux%20ci_yaml%20flutter%20roller

**Pull request merged**
